src/web/llvmir/lsp_model_provider.ts

```diff
@@ -124,7 +124,7 @@ export class LspModelProvider {
                             }
                         } else if (am.groups["user"] !== undefined) {
                             const varName = am.groups["user"];
-                            if (lastFunction?.values.get(varName) !== undefined) {
+                            if (varName.startsWith("%") && lastFunction !== undefined) {
```

Comment:
Do we handle globals whose initializer uses globals that are defined later?

Reply:
yes, a global will fail the `startsWith` check and so it will be added unconditionally to the users